BeagleBone_black/src/utils/math/kalman.hpp

```diff
@@ -28,39 +28,37 @@ namespace hyped {
 namespace utils {
 namespace math {
 
-// TODO(Uday): Fill in the class comment.
-// TODO(Uday): Explain a little bit about Kalman filters in class comment or file comment
 /**
- * @brief
+ * @brief    This class is for filtering the data from sensors to smoothen it.
  *
- * @tparam T
+ * @param[in] T    This is the type of the data value that it should filter.
  */
 template <typename T>
 class Kalman {
  public:
-  // TODO(Uday): Fill in the method comments.
   /**
-   * @brief Construct a new Kalman object
+   * @brief    Construct a new Kalman object
    *
-   * @param input_value
-   * @param measurement_noise
-   * @param process_noise
-   * @param estimation_error
+   * @param[in] input_value    Initial value of the reading usually set to 0
+   * @param[in] measurement_noise    Standard deviation of the measurement
+   * @param[in] process_noise    How noisy the measurement can be (predetermined)
+   * @param[in] estimation_error    Total error that has been collected over time usually set to 0
    */
-  Kalman(T input_value, T measurement_noise, T process_noise, T estimation_error = T());
+  Kalman(T measurement_noise, T process_noise, T input_value = T(), T estimation_error = T());
```

Comment:
Can you change this back please? Default constructor does not guarantee it will be zero so it is better to specify that explicitly. Hence there is no point in changing the interface and breaking code that already uses it.